packages/react/dropdown-menu/src/DropdownMenu.tsx

```diff
@@ -278,25 +281,16 @@ const DropdownMenuRootContent = React.forwardRef<
         hasInteractedOutsideRef.current = false;
       }}
       onInteractOutside={(event) => {
+        const target = event.target as HTMLElement;
+        const isTriggerTarget = !context.triggerRef.current?.contains(target);
         props.onInteractOutside?.(event);
 
         if (!event.defaultPrevented) {
           const originalEvent = event.detail.originalEvent as PointerEvent;
           const ctrlLeftClick = originalEvent.button === 0 && originalEvent.ctrlKey === true;
           const isRightClick = originalEvent.button === 2 || ctrlLeftClick;
-
-          if (!context.modal || isRightClick) hasInteractedOutsideRef.current = true;
```

Comment:
This was preventing the trigger from receiving focus when user clicks trigger to close because we have logic in `onCloseAutoFocus` that only focuses the trigger if it wasn't an outside interaction.

I have omitted the trigger from being considered "outside".